ta/storage/storage.c

```diff
@@ -662,3 +664,73 @@ TEE_Result ta_storage_cmd_get_obj_info(uint32_t param_types,
 
 	return res;
 }
+
+static TEE_Result clear_storage(uint32_t storage_id)
+{
+	TEE_ObjectEnumHandle oe = TEE_HANDLE_NULL;
+	TEE_Result enum_res = TEE_ERROR_GENERIC;
+	TEE_ObjectHandle o = TEE_HANDLE_NULL;
+	TEE_Result res = TEE_ERROR_GENERIC;
+	TEE_UUID uuid = TA_UUID;
+	TEE_ObjectInfo oi = { };
+	size_t obj_id_sz = 0;
+	void *obj_id = NULL;
+	size_t i = 0;
+
+	IMSG("Clearing TA storage (UUID: %pUl, storage ID: 0x%x)",
+	     (void *)&uuid, storage_id);
+	res = TEE_AllocatePersistentObjectEnumerator(&oe);
+	if (res)
+		return res;
+	res = TEE_StartPersistentObjectEnumerator(oe, storage_id);
+	if (res == TEE_ERROR_ITEM_NOT_FOUND) {
+		IMSG("No object found");
+		res = TEE_SUCCESS;
+		goto out;
+	}
+	if (res)
+		goto out;
+	obj_id = TEE_Malloc(TEE_OBJECT_ID_MAX_LEN, 0);
+	if (!obj_id)
+		goto out;
+
+	while (true) {
+		enum_res = TEE_GetNextPersistentObject(oe, &oi, obj_id,
+						       &obj_id_sz);
+		if (enum_res == TEE_ERROR_ITEM_NOT_FOUND)
+			break;
+		if (enum_res)
+			break;
```

Comment:
This case relates to an unexpected error. I think it should be reported by the TA as its storage may not have been wiped.

Reply:
Code updated to set `res = enum_res;`.